distributed/tests/test_worker.py

```diff
@@ -3462,3 +3487,139 @@ async def test_TaskState__to_dict(c, s, a):
     assert isinstance(tasks["z"], dict)
     assert tasks["x"]["dependents"] == ["<TaskState 'y' memory>"]
     assert tasks["y"]["dependencies"] == ["<TaskState 'x' memory>"]
+
+
+@gen_cluster(client=True)
+async def test_dups_in_pending_data_per_worker(c, s, a, b):
```

Comment:
I'm inclined to ditch this test. it was very helpful to reproduce and properly debug the previous issue but the various conditions leading up to the original `AssertionError` are now guaranteed to be impossible (e.g. uniqueness in heap). I kept it, for now, to "prove" that the previously borken workflow now succeeds.

Reply:
I agree, we should ditch it